cypress/integration/test_table.js

```diff
@@ -95,4 +95,86 @@ describe("tests table", function() {
       "_blank"
     );
   });
+
+  describe("Test Status Selector", () => {
+    beforeEach(() => {
+      cy.visit(TESTS_ROUTE);
+      cy.get("#cy-test-status-select > .cy-treeselect-bar").click();
+    });
+
+    it("Status select says 'No filters selected' by default", () => {
+      cy.get("#cy-test-status-select").contains("No filters selected");
+    });
+
+    it("Clicking on 'All' checkbox adds all statuses to URL ", () => {
+      cy.get(".cy-checkbox")
+        .contains("All")
+        .click();
+      cy.location().should(loc => {
+        expect(loc.pathname).to.equal(TESTS_ROUTE);
+        expect(loc.search).to.include("statuses=all,pass,fail,skip,silentfail");
+      });
+    });
+
+    const statuses = [
+      { display: "Pass", key: "pass" },
+      { display: "Silent Fail", key: "silentfail" },
+      { display: "Fail", key: "fail" },
+      { display: "Skip", key: "skip" }
+    ];
+
+    statuses.forEach(({ display, key }) => {
+      it(`Clicking on ${display} status checkbox adds ${key} status to URL and clicking again removes it`, () => {
+        cy.get(".cy-checkbox")
+          .contains(display)
+          .click();
+        cy.location().should(loc => {
+          expect(loc.pathname).to.equal(TESTS_ROUTE);
+          expect(loc.search).to.include(`statuses=${key}`);
```

Comment:
i think this test would still pass if clicking display added every status to url every time. should this test that loc.seach _only_ contains the selected status?

Reply:
sure